python/lsst/daf/butler/remote_butler/_registry.py

```diff
@@ -513,7 +513,19 @@ def queryDatasetAssociations(
         collectionTypes: Iterable[CollectionType] = CollectionType.all(),
         flattenChains: bool = False,
     ) -> Iterator[DatasetAssociation]:
-        raise NotImplementedError()
+        # queryCollections only accepts DatasetType.
+        if isinstance(datasetType, str):
+            datasetType = self.getDatasetType(datasetType)
+        resolved_collections = self.queryCollections(
+            collections, datasetType=datasetType, collectionTypes=collectionTypes, flattenChains=flattenChains
+        )
+        with self._butler._query() as query:
+            query = query.join_dataset_search(datasetType, resolved_collections)
+            result = query.general(
+                datasetType.dimensions,
+                dataset_fields={datasetType.name: {"dataset_id", "run", "collection", "timespan"}},
+            )
+            yield from DatasetAssociation.from_query_result(result, datasetType)
```

Comment:
@timj, this is how replacement code for `queryDatasetAssociations` is going to look like. I checked `RepoExportContext` class to see if I can replace `queryDatasetAssociations` usage there, but that class does not even have butler instance now, and it uses registry for everything. We can do it later and keep `queryDatasetAssociations` in registry for now. Both direct and remote registries use new query system.

Reply:
Great. I had forgotten that we need to change the export code so that it doesn't use registry.